lib/Marlin/Marlin/src/feature/prusa/MMU2/protocol_logic.cpp

```diff
@@ -758,7 +760,7 @@ void ProtocolLogic::LogRequestMsg(const uint8_t *txbuff, uint8_t size) {
     } else {
         MMU2::LogRequestMsg(tmp);
     }
-    strncpy(lastMsg, tmp, rqs);
+    strlcpy(lastMsg, tmp, rqs);
```

Comment:
There is no strlcpy on 8bit FW and this source file must remain 8bit compatible ... but I do agree that strncpy's behaviour is highly "unpredictable" so probably adding an explicit zero terminator into `lastMsg` would do the trick

Reply:
ok, will look into that.